File: 2265-partition-array-according-to-given-pivot/2265-partition-array-according-to-given-pivot.java
class Solution {
    public int[] pivotArray(int[] nums, int pivot) {
        
        List<Integer> larger = new ArrayList<Integer>();
        List<Integer> smaller = new ArrayList<Integer>();
        int pivotCount = 0;

        for(int i = 0; i < nums.length; i++){

            if(nums[i] < pivot){
                smaller.add(nums[i]);
            }

            else if(nums[i] > pivot){
                larger.add(nums[i]);
            }else pivotCount++;
        }

        List<Integer> newarr = new ArrayList<Integer>();

        while(pivotCount > 0){
            smaller.add(pivot);
            pivotCount--;
        }
        newarr.addAll(smaller);
        newarr.addAll(larger);

        System.out.println(newarr);

        return newarr.stream().mapToInt(i -> i).toArray();
    }
}

File: 2265-partition-array-according-to-given-pivot/2265-partition-array-according-to-given-pivot.kt
class Solution {
    fun pivotArray(nums: IntArray, pivot: Int): IntArray {
        val smallerList: MutableList<Int> = arrayListOf()
        val greaterList: MutableList<Int> = arrayListOf()
        var pivotCount = 0

        for (i in nums.indices) {
            if (nums[i] < pivot) {
                smallerList.add(nums[i])
            }else if (nums[i] > pivot) {
                greaterList.add(nums[i])
            }else pivotCount++
        }

        repeat(pivotCount){
            smallerList.add(pivot)
        }
        val mergedArray: IntArray = (smallerList + greaterList).toIntArray()

        return mergedArray
    }
}